src/main/java/com/olgaboiar/mint/IServer.java

```diff
@@ -2,13 +2,13 @@
 
 import java.io.IOException;
 
-public interface ServerInterface {
+public interface IServer {
     void start() throws IOException;
     void run() throws IOException;
     void acceptClientConnection() throws IOException;
     void listenToClientConnection() throws IOException;
     void readClientInput() throws IOException;
-    void sendResponseToClient(String[] responseArray) throws IOException;
+    void sendResponseToClient(Response response) throws IOException;
```

Comment:
I like this change, I think it helps to understand the responsibilities when talking in terms of domain, and using your own defined type rather than an array makes the interaction less fragile (i.e. accidentally changing the order somewhere upstream).